### PR TITLE
add virtual output referencs

### PR DESCRIPTION
**Pull request by amireluk**
NOTE: Microsft.DataCollaboration is in limited private preview.
We work closely with each customer and API breaks are still expected.

Changes:

- Library support to synapse pools
- Display name property for workspace
- Purpose property to script 
- Virtual output reference as new resource

NOTE: Microsft.DataCollaboration is in limited private preview.
We work closely with each customer and API breaks are still expected.

### CODE REVIEW

**Comment by erich-wang** [comment]
Adding `WaitForARMFeedback` as there is new api in this PR.

File: specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json [hunk 1]
```diff
@@ -637,6 +637,13 @@
               "$ref": "#/definitions/ConstrainedResource"
             }
           },
+          {
+            "name": "force",
+            "in": "query",
```
Comment (RamyasreeChakka) [inline specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json:642]: This is anti-pattern. Query parameters should be used in PUT requests.

File: specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json [hunk 2]
```diff
@@ -648,8 +655,8 @@
               "$ref": "#/definitions/ConstrainedResource"
             }
           },
-          "200": {
-            "description": "Success",
+          "202": {
```
Comment (RamyasreeChakka) [inline specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json:658]: We recommend to follow 200 + ProvisioningState pattern. Please see here: https://github.com/Azure/azure-resource-manager-rpc/blob/e64480d4f830bd3ac18286e4ee1b2769ca7283d4/v1.0/Addendum.md#creatingupdating-using-put

Comment (amireluk) [reply specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json:658]: The update operation is async as well, this pattern mean returning Azure-AsyncOperation as part of a 200 OK result ?

File: specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json (hunk 1 above)
Comment (amireluk) [reply specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json:642]: This API is following an existing Azure Synapse API that uses the same pattern:
https://docs.microsoft.com/en-us/rest/api/synapse/bigdatapools/createorupdate 

Calling PUT on this ConstrainedResource will effectively create a synapse spark pool by passing the request to Synapse.

So matching the existing API was the main reasoning here.
This is the same for the 201 accepted below. Synapse use this pattern and we align with it so we can return the same response.

File: specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json (hunk 2 above)
Comment (amireluk) [reply specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json:658]: (Also see comment above about API alignment)

**Comment by erich-wang** [comment]
@amireluk, please address ARM review comments, thanks.

**Comment by amireluk** [comment]
> @amireluk, please address ARM review comments, thanks.

I've addressed both comments. Waiting for ARM @RamyasreeChakka

**Comment by chiragg4u** [comment]
It looks like you’ve removed the auto checklist comment in the PR. The checklist is important and I would like to confirm if you’ve gone through the relevant documentation.

This is a mandatory requirement for the review. Please confirm the same for taking the process forward.

**Comment by amireluk** [comment]
@chiragg4u  - I've gone through the checklist before, and revisited it again now.
If there anything we're doing that's not according to guidelines please lmk.

Adding the checklist here for reference:

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://aka.ms/SwaggerPRReview).

**Comment by amireluk** [comment]
[Pasting here full checklist]

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

**Comment by j5lim** [comment]
>     "/providers/Microsoft.DataCollaboration/listInvitations": {

The example indicates the routing is location based. It should be like /providers/Microsoft.DataCollaboration/locations/{location}/consumerInvitations to be consistent with the a point GET API below.

---
Refers to: specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json:19 in 0fca625. [](commit_id = 0fca6258194341acd75075be0471a318e77277ed, deletion_comment = False)

**Comment by j5lim** [comment]
>         },

Proxy resource can't have a top level location property. ARM will use the tracked parent resource's location if the api is location-based.

---
Refers to: specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json:5194 in 0fca625. [](commit_id = 0fca6258194341acd75075be0471a318e77277ed, deletion_comment = False)

**Comment by j5lim** [comment]
>       "headers": {

Since the api is async, the response should contain provisioningState to indicate the status of create operation.
[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#provisioningstate-property](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#provisioningstate-property)

It should also contain Azure-asyncOperation header to allow polling the async operation status.
[https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc0025-201-is-the-recommended-async-pattern](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc0025-201-is-the-recommended-async-pattern)

---
Refers to: specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/examples/Workspaces_CreateOrUpdate.json:23 in 0fca625. [](commit_id = 0fca6258194341acd75075be0471a318e77277ed, deletion_comment = False)

File: specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json (hunk 2 above)
Comment (j5lim) [inline specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json:658]: >202": { [](start = 11, length = 7)

Please follow the pattern in RPC. [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creatingupdating-using-put)

**Comment by erich-wang** [comment]
@amireluk, could you address comments?

**Comment by erich-wang** [comment]
@amireluk, is this PR still valid, we'll close this PR if there's no response from your side by end of April 7, thanks for your understanding.

**Comment by erich-wang** [comment]
@amireluk , I'm closing this PR as there's no response from you for almost one month. Please create a new PR or reopen this PR if necessary.

**Comment by amireluk** [comment]
> @amireluk , I'm closing this PR as there's no response from you for almost one month. Please create a new PR or reopen this PR if necessary.

I think this will be best, we introduced more changes during the past few weeks and are expecting even more to come.
 Will make more sense to apply them together.
 
 Ill close this PR for the time being.

**Comment by JackTn** [comment]
Hi, @amireluk. The PR has be closed for a long time and it's related branch still exist. Please tell me if you still need this branch or i will delete it in 14 days.

**Pull request closed without merging**